src/checkout.c

```diff
@@ -1371,39 +1372,37 @@ static int mkpath2file(
 	return error;
 }
 
-static int buffer_to_file(
-	checkout_data *data,
-	struct stat *st,
-	git_buf *buf,
-	const char *path,
-	mode_t file_mode)
-{
-	int error;
-
-	if ((error = mkpath2file(data, path, data->opts.dir_mode)) < 0)
-		return error;
+struct checkout_stream {
+	git_filter_stream base;
```

Comment:
Just a nitpick, but we do have a preference for calling this 'parent', rather than base (I think it's just the iterators which use this name).